File: php/src/org/shypl/sna/impl/OkSocialNetwork.php
<?php

namespace org\shypl\sna\impl;

use org\shypl\sna\Adapter;
use org\shypl\sna\SocialNetwork;

final class OkSocialNetwork extends SocialNetwork {

	/**
	 *
	 */
	public function __construct() {
		parent::__construct(3, "ok");
	}

	/**
	 * @param array $parameters
	 *
	 * @return Adapter
	 */
	public function createAdapter(array $parameters) {
		return new OkAdapter($parameters);
	}
}

File: java/src/main/java/org/shypl/sna/impl/OkSocialNetwork.java
package org.shypl.sna.impl;

import org.shypl.common.util.DeclinationImpl;
import org.shypl.sna.Currency;
import org.shypl.sna.SocialNetwork;

public final class OkSocialNetwork extends SocialNetwork {
	public OkSocialNetwork() {
		super(3, "ok", "Одноклассники", new Currency("ок", new DeclinationImpl("ок", "ок", "ок")));
	}
}
